core/client/fs/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -90,16 +97,33 @@
   /** Whether to cache blocks in this file into Alluxio. */
   private final boolean mShouldCache;
 
-  // The following 3 fields must be kept in sync. They are only updated in updateStreams together.
+  // The following 5 fields must be kept in sync. They are only updated together in
```

Comment:
I'm inclined to prefer some sort of "state" object, which will take care of keeping all of these fields in sync. It would reduce the possibility of introducing a sync bug. What do you think?